src/main/java/Domain/ExternalServices/PaymentService/AdapterPayment.java

```diff
@@ -1,15 +1,16 @@
 package Domain.ExternalServices.PaymentService;
+import java.lang.reflect.Proxy;
 import java.util.logging.Logger;
 import Domain.ExternalServices.ExternalService;
 
 // TODO: Implement the Adapter pattern
 public class AdapterPayment implements ExternalService {
 
+    private ProxyPayment _paymentService;
```

Comment:
here you keep the interface. and you give the proxy in the construdtor when you initiate the class.
we dont have a real class anyway so i think its fine like that